quantum/encoder.h

```diff
@@ -34,48 +47,48 @@ bool encoder_dequeue_event(uint8_t *index, bool *clockwise);
 bool encoder_update_kb(uint8_t index, bool clockwise);
 bool encoder_update_user(uint8_t index, bool clockwise);
 
-#    ifdef SPLIT_KEYBOARD
-
-#        if defined(ENCODERS_PAD_A_RIGHT)
-#            ifndef NUM_ENCODERS_LEFT
-#                define NUM_ENCODERS_LEFT ARRAY_SIZE(((pin_t[])ENCODERS_PAD_A))
-#            endif
-#            ifndef NUM_ENCODERS_RIGHT
-#                define NUM_ENCODERS_RIGHT ARRAY_SIZE(((pin_t[])ENCODERS_PAD_A_RIGHT))
-#            endif
-#        else
-#            ifndef NUM_ENCODERS_LEFT
-#                define NUM_ENCODERS_LEFT ARRAY_SIZE(((pin_t[])ENCODERS_PAD_A))
-#            endif
-#            ifndef NUM_ENCODERS_RIGHT
-#                define NUM_ENCODERS_RIGHT NUM_ENCODERS_LEFT
-#            endif
+#ifdef SPLIT_KEYBOARD
+
+#    if defined(ENCODER_A_PINS_RIGHT)
+#        ifndef NUM_ENCODERS_LEFT
+#            define NUM_ENCODERS_LEFT ARRAY_SIZE(((pin_t[])ENCODER_A_PINS))
 #        endif
-#        ifndef NUM_ENCODERS
-#            define NUM_ENCODERS (NUM_ENCODERS_LEFT + NUM_ENCODERS_RIGHT)
+#        ifndef NUM_ENCODERS_RIGHT
+#            define NUM_ENCODERS_RIGHT ARRAY_SIZE(((pin_t[])ENCODER_A_PINS_RIGHT))
 #        endif
-
-#    else // SPLIT_KEYBOARD
-
-#        ifndef NUM_ENCODERS
-#            define NUM_ENCODERS ARRAY_SIZE(((pin_t[])ENCODERS_PAD_A))
+#    else
+#        ifndef NUM_ENCODERS_LEFT
+#            define NUM_ENCODERS_LEFT ARRAY_SIZE(((pin_t[])ENCODER_A_PINS))
+#        endif
+#        ifndef NUM_ENCODERS_RIGHT
+#            define NUM_ENCODERS_RIGHT NUM_ENCODERS_LEFT
 #        endif
-#        define NUM_ENCODERS_LEFT NUM_ENCODERS
-#        define NUM_ENCODERS_RIGHT 0
+#    endif
+#    ifndef NUM_ENCODERS
+#        define NUM_ENCODERS (NUM_ENCODERS_LEFT + NUM_ENCODERS_RIGHT)
+#    endif
 
-#    endif // SPLIT_KEYBOARD
+#else // SPLIT_KEYBOARD
 
 #    ifndef NUM_ENCODERS
-#        define NUM_ENCODERS 0
-#        define NUM_ENCODERS_LEFT 0
-#        define NUM_ENCODERS_RIGHT 0
-#    endif // NUM_ENCODERS
+#        define NUM_ENCODERS ARRAY_SIZE(((pin_t[])ENCODER_A_PINS))
+#    endif
+#    define NUM_ENCODERS_LEFT NUM_ENCODERS
+#    define NUM_ENCODERS_RIGHT 0
+
+#endif // SPLIT_KEYBOARD
 
-#    define NUM_ENCODERS_MAX_PER_SIDE MAX(NUM_ENCODERS_LEFT, NUM_ENCODERS_RIGHT)
+#ifndef NUM_ENCODERS
+#    define NUM_ENCODERS 0
+#    define NUM_ENCODERS_LEFT 0
+#    define NUM_ENCODERS_RIGHT 0
+#endif // NUM_ENCODERS
```

Comment:
Also looks like this block is (and was) useless, because both branches of `#ifdef SPLIT_KEYBOARD` above end up defining `NUM_ENCODERS`.